engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -164,15 +267,23 @@ Object doWarning(
       }
     }
 
-    arguments[thisArgumentPosition] = self.getValue();
-    ArrayRope<Warning> warnings = self.getReassignedWarnings(this);
-    Object result = childDispatch.execute(frame, state, symbol, self.getValue(), arguments);
-    return WithWarnings.prependTo(result, warnings);
+    Warning[] arrOfWarnings;
+    try {
+      arrOfWarnings = warnings.getWarnings(self, this);
+    } catch (UnsupportedMessageException e) {
+      // Not possible due to `hasWarnings` check
+      throw new IllegalStateException(e);
```

Comment:
`shouldNotReachHere` allows one to add the comment as a string to parameter to the method.

Reply:
We are not particularly consistent on that and throw `IllegalStateException` on such occasions. But sure, I can change it